### PR TITLE
Animation packet, upgrades to equip_item

### PR DESCRIPTION
**Pull request by iT0xic**
Added Animation handling (0x781)
Changed computed_values::runSpeed default to CONST
Added secondary removal when trying to equip 2H weapon

---
This change is [Reviewable](https://reviewable.io/reviews/dev-osrose/osirose-new/235)

### CODE REVIEW

File: src/map/src/combat/player.cpp [hunk 1]
```diff
@@ -74,3 +76,12 @@ void Player::toggle_player_move(EntitySystem& entitySystem, RoseCommon::Entity e
 	pToggle.set_object_id(basicInfo.id);
 	entitySystem.send_map(pToggle);
 }
+
+void Player::set_animation(EntitySystem& entitySystem, RoseCommon::Entity entity, const RoseCommon::Packet::CliSetAnimation& packet) {
+	auto& basicInfo = entitySystem.get_component<Component::BasicInfo>(entity);
```
Comment (L3nn0x) [inline src/map/src/combat/player.cpp]: Can we use const here? There is no reason to have this non-const

File: src/map/src/combat/player.cpp [hunk 2]
```diff
@@ -74,3 +76,12 @@ void Player::toggle_player_move(EntitySystem& entitySystem, RoseCommon::Entity e
...
+	auto pAnimation = Packet::SrvSetAnimation::create();
+	pAnimation.set_id(packet.get_id());
+	pAnimation.set_value(packet.get_value());
+	pAnimation.set_object_id(basicInfo.id);
+	entitySystem.send_nearby(entity, pAnimation);
```
Comment (L3nn0x) [inline src/map/src/combat/player.cpp:86]: We will probably want to add some checks here at some point, to make sure the clients have the animation / are not trying to hack

File: src/map/src/items/inventory.cpp [hunk 3]
```diff
@@ -52,6 +52,10 @@ inline bool is_spot_correct(const EntitySystem& entitySystem, RoseCommon::Entity
     }
 }
 
+inline bool is_two_handed(uint16_t subtype) {
+    return ((subtype >= ItemSubType::TWOH_SWORD && subtype <= ItemSubType::DUAL_WEILD) && (subtype != ItemSubType::WAND));
```
Comment (L3nn0x) [inline src/map/src/items/inventory.cpp]: Can we change `DUAL_WEILD` to `DUAL_WIELD` here? (And anywhere else where it's relevant)

File: src/rosecommon/include/enums.h [hunk 4]
```diff
@@ -17,6 +17,22 @@ enum RidingItem : uint8_t {
...
+    SPEAR = 222,
+    TWOH_AXE = 223,
+    BOW = 231,
+    GUN,
+    LAUNCHER,
+    STAFF = 241,
+    WAND,
+    KATAR = 251,
+    DUAL_WEILD,
+    XBOW = 271
```
Comment (L3nn0x) [inline src/rosecommon/include/enums.h:33]: What's the difference between a bow and an xbow?

**Review by L3nn0x — COMMENTED** [review]
rest lgtm!

File: src/rosecommon/include/enums.h (hunk 4 above)
Comment (iT0xic) [reply src/rosecommon/include/enums.h:33]: They have a different subtype.
More over, bow = 2h-weap, xbow = 1h-weap

File: src/map/src/combat/player.cpp (hunk 2 above)
Comment (iT0xic) [reply src/map/src/combat/player.cpp:86]: Client is handling all those checks. In case there is no motion it just doing nothing :)

**Review by L3nn0x — APPROVED** [review]

**Pull request merged**